From 2141110771b4600fdee4e1f79ad2d792afb3ce0a Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <heinrich.schuchardt@canonical.com>
Date: Fri, 13 Aug 2021 14:35:02 +0200
Subject: [PATCH v2 1/1] libgcrypt: avoid -Wsign-compare in rijndael do_setkey
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Avoid a warning

lib/libgcrypt-grub/cipher/rijndael.c:352:21: warning:
comparison of integer expressions of different signedness:
‘int’ and ‘unsigned int’ [-Wsign-compare]
  352 |       for (i = 0; i < keylen; i++)
      |

Signed-off-by: Heinrich Schuchardt <heinrich.schuchardt@canonical.com>
Reviewed-by: Daniel Kiper <daniel.kiper@oracle.com>
---
v2:
	use 'unsigned int' instead of 'unsigned'
---
 grub-core/lib/libgcrypt/cipher/rijndael.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/grub-core/lib/libgcrypt/cipher/rijndael.c b/grub-core/lib/libgcrypt/cipher/rijndael.c
index 9d884c9d3..acb6ca66e 100644
--- a/grub-core/lib/libgcrypt/cipher/rijndael.c
+++ b/grub-core/lib/libgcrypt/cipher/rijndael.c
@@ -181,7 +181,8 @@ do_setkey (RIJNDAEL_context *ctx, const byte *key, const unsigned keylen)
   static int initialized = 0;
   static const char *selftest_failed=0;
   int rounds;
-  int i,j, r, t, rconpointer = 0;
+  unsigned int i;
+  int j, r, t, rconpointer = 0;
   int KC;
   union
   {
-- 
2.30.2

